lib/Rx/React/Promise.php

```diff
@@ -76,7 +76,10 @@ function ($value) use ($subject) {
                         $subject->onNext($value);
                         $subject->onCompleted();
                     },
-                    [$subject, "onError"]
+                    function ($error) use ($subject) {
+                        $error = $error instanceof \Exception ? $error : new RejectedPromiseException($error);
```

Comment:
I had considered doing that, but I thought it better to not wrap values that are already exceptions as I consider a non-exception value a special case. @asm89 - do you have any thoughts on this?